src/Terminus.php

```diff
@@ -100,7 +101,11 @@ public function __construct(Config $config, InputInterface $input, OutputInterfa
         $this->addBuiltInCommandsAndHooks();
         $this->addPluginsCommandsAndHooks();
 
-        $application->addSelfUpdateCommand('pantheon-systems/terminus');
+        // We can't use Robo\Application addSelfUpdateCommand because if plugin manager is running it won't be a phar from there.
```

Comment:
So you are saying the `self:plugin:update` command is interfering with `consolidation/self-update` (a dependency of `consolidation/robo`?

Reply:
Nope. This check from robo: https://github.com/consolidation/robo/blob/3.x/src/Application.php#L71 was returning FALSE (Phar::running) and therefore not adding the self update command when invoked from the build dependencies folder

Reply:
Gotcha.

Reply:
Should we add a test to confirm `self:update` exists in the phar?

Reply:
I'll create a new PR with this test